packages/gatsby/src/utils/page-data.js

```diff
@@ -35,7 +40,71 @@ const write = async ({ publicDir }, page, result) => {
   await fs.outputFile(filePath, bodyStr)
 }
 
+const getChangedPageDataKeys = (state, cachedPageData) => {
+  if (cachedPageData && state.pageData) {
+    const pageKeys = []
+    state.pageData.forEach((value, key) => {
+      if (!cachedPageData.has(key)) {
+        pageKeys.push(key)
+      } else {
+        const newPageData = JSON.stringify(value)
+        const previousPageData = JSON.stringify(cachedPageData.get(key))
```

Comment:
we no longer need to `JSON.stringify` those 2 because it's already hash (string)

Reply:
Done 🙇  @pieh 
Also done a bit of renaming to make it clearer that we are comparing the resultHash